app/controllers/health_profiles_controller.rb

```diff
@@ -1,48 +1,58 @@
 class HealthProfilesController < ApplicationController
 	before_action :require_login
+ health_profile
+
 	
+ master
 
 	def index
 	end
 
 	def new
+ 
+		@patient = Patient.find(params[:patient_id])
 		# @health_profile = current_user.patients.find(params[:id])
 		@health_profile = HealthProfile.new
 	end
 
 	def create
-		# @health_profile = current_user.health_profiles.new(health_profile_params)
-		@health_profile = HealthProfile.new
+health_profile
+		@patient = Patient.find(params[:patient_id])
+		@health_profile = HealthProfile.new(health_profile_params)
+		@health_profile.patient_id = current_user.id
 		if @health_profile.save
-			redirect_to health_profile_path, notice: "Your profile has been created."
+			redirect_to patient_health_profile_path(@patient.id, @health_profile.id), notice: "Your profile has been created."
 		else
-			flash[:notice] = "Error creating new profile."
-			render 'new'
+			redirect_to new_patient_health_profile, notice: "Create profile unsuccessful. Please try again."
+
 		end
 
 	end
 
 	def show
-		# @health_profile = current_user.health_profiles.find(params[:id])
+    health_profile
 		@health_profile = HealthProfile.find(params[:id])
-		@allergies = health_profile.allergies.find(params[:patient_id])
-		@immunizations = health_profile.immunizations.find(params[:patient_id])
-		@prescriptions = health_profile.prescriptions.find(params[])
```

Comment:
Are we taking this info out of the health profile or is it going to be added later?